### PR TITLE
[MDETR] Directly use module dict for VQA heads

### PR DESCRIPTION
**Pull request by ebsmothers**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #246
* __->__ #232
* #231
* #230

Summary:
Remove the MultiHead class used for VQA heads, which is just a wrapper around ModuleDict anyways.

Test plan:

```
CUBLAS_WORKSPACE_CONFIG=:4096:8 python -m torch.distributed.launch --nproc_per_node=4 --use_env vqa_eval.py
--resume https://pytorch.s3.amazonaws.com/models/multimodal/mdetr/gqa_resnet101_checkpoint.pth --ema --eval
--dataset_config vqa.json
```
Gives total accuracy 61.6% as expected

Differential Revision: [D38467123](https://our.internmc.facebook.com/intern/diff/D38467123)

### CODE REVIEW

**Review by RdoubleA — APPROVED** [review]

File: torchmultimodal/models/mdetr/model.py [hunk 1]
```diff
@@ -384,7 +369,7 @@ def mdetr_for_vqa(
 ) -> MDETRForVQA:
     hidden_dim = transformer_d_model
     vqa_heads = mdetr_gqa_heads()
-    num_heads = len(vqa_heads.heads.keys())
+    num_heads = len(vqa_heads.keys())
```
Comment (ankitade) [inline torchmultimodal/models/mdetr/model.py:372]: just len(vqa_heads) should work?

**Review by ankitade — APPROVED** [review]
this answers my question on previous PR

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38467123).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38467123).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38467123).

**Comment by ebsmothers** [comment]
@ebsmothers has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38467123).

**Pull request closed without merging**